### PR TITLE
#9176 Fix inconsistent behavior of compile time macros when assigned to a constant

### PR DESCRIPTION
**Pull request by ianmcxa**
Very simple change marking gorge, gorgeEx and staticExec as `.compileTime.` to fix inconsistent behavior when setting the result of gorgeEx to a constant.

### CODE REVIEW

**Comment by timotheecour** [comment]
great! 

- [ ] do you know if there more cases like this? 
ie, what @LemonBoy described here:
> Ehh, it's a stupid bug in the implementation of some compile-time only procs.

- [ ] a test would be needed I think, eg maybe something like:
```nim
when defined(posix):
  const cmd = "echo foo" # or if you have another idea that makes it clear it's being run at CT, not at RT; but at least it'll catch the empty output bug of https://github.com/nim-lang/Nim/issues/9176
  var ret1 = staticExec(cmd)
  const tmp = staticExec(cmd)
  var ret2 = tmp
  doAssert ret1 == ret2
```

**Comment by ianmcxa** [comment]
Should I make a test for each of the commands?

**Comment by timotheecour** [comment]
meh, at least 1 is a good start

**Comment by krux02** [comment]
Interesting diff, I like it. Please add a test case that fails on the old behavior and works correctly on the new behavior.

**Comment by timotheecour** [comment]
> I've gone ahead and added a fix for getProjectPath
maybe update PR title

**Comment by Clyybber** [comment]
@dom96 I think this is an acceptable tradeoff, because we have "static" in the name of the proc (ignoring gorge, since that alias is wierd anyways imo).

**Comment by ianmcxa** [comment]
Why do we have gorge and gorgeEx anyway?

**Comment by skellock** [comment]
`gorgeEx` gives you the exit code in addition to the `stdout`/`stderr`.

`slurp` and `gorge` are odd names for sure. I do prefer the much more boring static* nomenclature.

**Comment by ianmcxa** [comment]
My question was just why do we have that naming convention in the first place? Why not a single proc that does what gorgeEx does now?

**Comment by skellock** [comment]
Not sure of the heritage, but I like where you're going.  #1994 is distantly related if we're talking about tidying up the API before 1.0.

**Comment by Araq** [comment]
> Why do we have gorge and gorgeEx anyway?

Because some young fool (@Araq) had fun designing the language and added a single bad joke.

File: lib/system.nim [hunk 1]
```diff
@@ -3789,12 +3789,10 @@ proc staticRead*(filename: string): string {.magic: "Slurp".}
   ##
   ## `slurp <#slurp>`_ is an alias for ``staticRead``.
 
-proc gorge*(command: string, input = "", cache = ""): string {.
-  magic: "StaticExec".} = discard
```
Comment (Araq) [inline lib/system.nim]: Removing this ``magic`` here cannot be right.

File: lib/system.nim [hunk 2]
```diff
@@ -3789,12 +3789,10 @@ proc staticRead*(filename: string): string {.magic: "Slurp".}
...
+proc gorge*(command: string, input = "", cache = ""): string {.compileTime.} = discard
   ## This is an alias for `staticExec <#staticExec>`_.
 
-proc staticExec*(command: string, input = "", cache = ""): string {.
-  magic: "StaticExec".} = discard
```
Comment (Araq) [inline lib/system.nim]: Removing this ``magic`` here cannot be right.

**Comment by Araq** [comment]
This needs to be solved differently.

**Pull request closed without merging**